cockpit-container-update

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Update .cockpit-ci/container to the latest tasks container tag automatically
+
+import json
+import os
+import socket
+import sys
+import urllib.parse
+import urllib.request
+from ssl import SSLEOFError
+
+import task
+from lib.constants import BASE_DIR
+
+sys.dont_write_bytecode = True
+
+
+URL = 'https://quay.io/api/v1/repository/cockpit/tasks/tag/?onlyActiveTags=true'
+
+
+def fetch_tags(url):
+    try:
+        req = urllib.request.Request(url)
+        with urllib.request.urlopen(req) as response:
+            return json.loads(response.read().decode().strip())
+    except (urllib.error.HTPPError, ConnectionResetError, socket.error, SSLEOFError) as e:
+        print("Transient error during quay.io request %s", e)
+
+    return None
+
+
+def run(context, verbose=False, **kwargs):
+    container = context or os.path.join('.cockpit-ci', 'container')
+    container_path = os.path.join(BASE_DIR, container)
+
+    with open(container_path) as fp:
+        content = fp.read()
+
+    current_tag = content.strip().partition(':')[-1]
+    tags = fetch_tags(URL)
+    if tags is None:
+        return
```

Comment:
In theory, projects could also use a different container. Let's not blindly overwrite that with quay/tasks, but actually read the image name and put it into the URL. And exit with a message if it doesn't start with `quay.io`?